libraries/ESP8266WebServer/src/ESP8266WebServer.h

```diff
@@ -231,13 +231,11 @@ class ESP8266WebServerTemplate
 
 };
 
+};
 
 #include "ESP8266WebServer-impl.h"
 #include "Parsing-impl.h"
```

Comment:
These two were inside the namespace because their implementations are _not_ in a namespace. Pulling them out like this means that the internals are moved to the global namespace, which is undesirable.

Reply:
the issue is that the things that they #include'ed was now being put into that namespace, which causes duplicate instanciations (code bloat) or missing symbols..

Reply:
You're correct about the last part. In general, it is bad practice to include inside a namespace for this exact reason.
The correct way to handle this is to separate the headers, so a bit more work along the lines of the following. Are you game?
1. ESP8266WebServer-impl.h: 
    * Put `namespace esp8266webserver` around all the code in there. 
    * In ESP8266WebServer.h move the include to after the namespace closing like you had previously proposed.
    * Move the three headers added to ESP8266WebServer.h in your latest proposal back to the top of ESP8266WebServer-impl.h.
2. RequestHandler.h: Wrapping with a namespace here could cause havoc in multiple dimensions. To avoid that: 
    * Fully qualify the template import like so: `using WebServerType = esp8266webserver::ESP8266WebServerTemplate<ServerType>;`
3. RequestHandlersImpl.h:
    * Keep your removal of `using namespace mime` in the global namespace. and its use where specifically used
    * Qualify the using WebServerType same way as above (x2)
 4. Move the above includes out of the esp8266 namespace as you had proposed.

Reply:
I can do that cleanup. does it belong into this PR? I would prefer it as a followup.

Reply:
Well, I proposed the above as part of this PR because you started the include changes here. If you prefer to separate, what you could do is keep just the integer overflow fix for the ::authenticate() method here, and do the include cleanup in a different PR. What do you think?